From: Stephane Glondu <steph@glondu.net>
Date: Sat, 2 Jul 2011 17:56:41 +0200
Subject: Add missing -cclib in ocamlc -a

This allows simple linking with -custom by naming only .cma files
(used by advi).

Remark: runtime support libraries (dll*.so, lib*.a) and the OCaml
archives (*.cma, *.cmxa) are built in an ad-hoc way using ocamlmklib,
ocamlc -a and ocamlopt -a. It would be better (but more intrusive) to
use only ocamlmklib.

Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 src/OMakefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/OMakefile b/src/OMakefile
index e198fa1..013164f 100644
--- a/src/OMakefile
+++ b/src/OMakefile
@@ -225,7 +225,7 @@ MakeLibOCaml(name) =
     $(LIBRARY_PREFIX)_$(name).cma $(LIBRARY_PREFIX)_$(name).cmxa: $(MakeLibC $(name))
     # OCAML_{BYTE,NATIVE}_LINK_FLAGS do not work for OCamlLibrary
     # Bit dirty workaround
-    OCAMLCFLAGS+=-dllib -l$(LIBRARY_PREFIX)_$(name)
+    OCAMLCFLAGS+=-dllib -l$(LIBRARY_PREFIX)_$(name) -cclib -l$(LIBRARY_PREFIX)_$(name)
     OCAMLOPTFLAGS+=-cclib -l$(LIBRARY_PREFIX)_$(name)
     # use mapprefix instead of addprefix, otherwise "-cclib -lgif"
     # is considered one option.
-- 
